src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/applicationinfo.cpp

```diff
@@ -240,15 +240,15 @@ APPLICATION_INFO::ShutDownApplication(const bool fServerInitiated)
             return;
         }
         app = m_pApplication.get();
-    }
-
-    LOG_INFOF(L"Stopping application '%ls'", QueryApplicationInfoKey().c_str());
-    app->Stop(fServerInitiated);
 
-    SRWExclusiveLock lock(m_applicationLock);
```

Comment:
Why was this lock here before?

Reply:
I think you fell for the same trap Aditya did, the code was moved into the first lock, have to pay close attention to the braces.